Frontend/ui-driver/src/Screens/Types.purs

```diff
@@ -2110,4 +2124,126 @@ type WelcomeScreenState = {
+  coinHistoryItems :: Array CoinHistoryItem,
+  usageHistoryItems :: Array CoinHistoryItem,
+  coinsEarnedToday :: Int,
+  expiringCoins :: Int,
+  expiringDays :: Int,
+  hasActivePlan :: Boolean,
+  timerID :: String,
+  timer :: Int,
+  totalCoinConvertedToCash :: Number,
+  coinConvertedToCashUsedForLatestDues :: Maybe Int,
+  coinConvertedTocashLeft :: Number,
+  coinConversionRate :: Number,
+  coinsToUse :: Int,
+  config :: AppConfig,
+  earningHistoryItems :: Array CoinHistoryItem,
+  weeklyEarningData :: Array WeeklyEarning,
+  anyRidesAssignedEver :: Boolean
+}
+
+type DriverEarningsScreenProps = {
+  subView :: DriverEarningsSubView,
+  date :: String,
+  popupType :: DriverEarningsPopupType,
+  showCoinsRedeemedAnim :: String,
+  calendarState :: CalendarState,
+  showCoinsUsagePopup :: Boolean,
+  selectedBarIndex :: Int,
+  weekIndex :: Int,
+  totalEarningsData :: TotalEarningsData,
+  currWeekData :: Array WeeklyEarning,
+  weekDay :: Array String,
+  currentWeekMaxEarning :: Int,
+  showShimmer :: Boolean,
+  startDate :: String,
+  endDate :: String,
+  gotDataforWeek :: Array Boolean,
+  coinConvertedSuccess :: Boolean,
+  individualQuestion :: FaqQuestions
+}
+
+type CalendarState = { 
+  calendarPopup :: Boolean,
+  endDate :: Maybe Common.CalendarModalDateObject,
+  selectedTimeSpan :: Common.CalendarModalDateObject,
+  startDate :: Maybe Common.CalendarModalDateObject,
+  weeks  :: Array Common.CalendarModalWeekObject
+}
+
+type FaqQuestions = {
+  question :: String,
+  videoLink :: Maybe String,
+  answer :: Array String,
+  showTable :: Boolean
+}
+type WeeklyEarning = {
+  earnings :: Int,
+  rideDistance :: Int,
+  rideDate :: String,
+  noOfRides :: Int,
+  percentLength :: Number
+}
+
+type TotalEarningsData = {
+  fromDate :: String,
+  toDate :: String,
+  totalEarnings :: Int,
+  totalRides :: Int,
+  totalDistanceTravelled :: Int
+}
+
+newtype CachedEarningsForDriver = CachedEarningsForDriver {
+  id :: String,
+  earningsData :: Array WeeklyEarning
+}
+
+derive instance genericCachedEarningsForDriver :: Generic CachedEarningsForDriver _
+derive instance newtypeCachedEarningsForDriver :: Newtype CachedEarningsForDriver _
+instance showCachedEarningsForDriver :: Show CachedEarningsForDriver where show = genericShow
+instance decodeCachedEarningsForDriver :: Decode CachedEarningsForDriver where decode = defaultDecode
+instance encodeCachedEarningsForDriver :: Encode CachedEarningsForDriver where encode = defaultEncode
+
+data DriverEarningsSubView = EARNINGS_VIEW | YATRI_COINS_VIEW | USE_COINS_VIEW | FAQ_VIEW | FAQ_QUESTON_VIEW
+
+derive instance genericDriverEarningsSubView :: Generic DriverEarningsSubView _
+instance showDriverEarningsSubView :: Show DriverEarningsSubView where show = genericShow
+instance eqDriverEarningsSubView :: Eq DriverEarningsSubView where eq = genericEq
+instance decodeDriverEarningsSubView :: Decode DriverEarningsSubView where decode = defaultEnumDecode
```

Comment:
do we need the decode and encode instances?

Reply:
yes